Add basic support for 128 players in scoreboard and spectator UI

Robyt3 opened the pull request:
Support showing up to 128 players in the scoreboard and the spectator UI. The 128 players are rendered in 3 columns of 43 players in the scoreboard and in 4 columns of 32 players in the spectator UI. Other lists like the server info (ingame and in the server browser) and the voting list already support showing any number of players.

Screenshots:

- ![screenshot_2024-07-15_23-28-30](https://github.com/user-attachments/assets/30b3321f-8b9f-4e32-9c1b-28a275678ef6)
- ![screenshot_2024-07-14_20-52-40](https://github.com/user-attachments/assets/7ce9e14c-d84e-48b0-9d31-93cb16d11b10)

## Checklist

- [X] Tested the change ingame
- [X] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test (especially base/) or added coverage to integration test
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

furo321 commented:
A bit unrelated, but there is also the stats board (`+statboard`). Which is limited to 32 players, but might be hard to make it display more than that.

https://github.com/ddnet/ddnet/blob/b0f0292fc9706dad6baa18e2161813a48d4202d7/src/game/client/components/statboard.cpp#L179-L182

Robyt3 commented:
> A bit unrelated, but there is also the stats board (`+statboard`). Which is limited to 32 players, but might be hard to make it display more than that.

TBH, I'm not entirely sure about the purpose of the statboard as it is. I don't think you would/could quickly look up any of the information in the statboard while actively playing especially with many players. It seems more appropriate to me to put this information on a separate menu page in a listbox.

heinrich5991 commented:
Perhaps the 128 player list could use 3 or even 4 columns? At this text size, it looks like there's more horizontal space.

Robyt3 commented:
> Perhaps the 128 player list could use 3 or even 4 columns? At this text size, it looks like there's more horizontal space.

There wouldn't be any space for the clan with 4 columns even with the smallest font size:

![screenshot_2024-07-15_13-24-47](https://github.com/user-attachments/assets/b6c068da-1d7f-46f4-994b-37ad4c8b084c)

![screenshot_2024-07-15_13-29-16](https://github.com/user-attachments/assets/097d36c2-397f-4d34-b394-502cd787d6b2)

3 columns could be possible but clans might still get truncated:

![screenshot_2024-07-15_13-23-31](https://github.com/user-attachments/assets/2f24d8f7-0cca-45d6-9e41-6d8a7ba016c4)

heinrich5991 commented:
I think the three columns would work better than two columns, because the text is slightly larger. I don't think a dropped character from a player nickname or clan is that bad.

Robyt3 commented:
Which one do you prefer: font size 12 or 13?

Font size 12: ![screenshot_2024-07-15_17-46-43](https://github.com/user-attachments/assets/e085beb8-9686-4aa7-a7fc-fe44314fbaa0)

Font size 13: ![screenshot_2024-07-15_17-47-18](https://github.com/user-attachments/assets/3b15f2df-f1bf-456d-8b87-7f34f81aa0df)

def- commented:
13-14 with smaller time and clan fields

Robyt3 commented:
Changed it to 3 columns of 43 players with font size 14 and removed some space for the time, ping and clan:

![screenshot_2024-07-15_23-28-30](https://github.com/user-attachments/assets/30b3321f-8b9f-4e32-9c1b-28a275678ef6)

heinrich5991 reviewed (approved):
Thanks a lot!

def- commented:
I'm wondering if we should put this into 18.4, probably let's not rush it.

heinrich5991 commented:
This is currently dead code (the client won't allow the server to send 128 players), so it's meaningless for the release.
